### PR TITLE
[player] Unregister from DBus upon destruction. Fixes MER#1350

### PR DESCRIPTION
**Pull request by tanty**

### CODE REVIEW

**Comment by tanty** [comment]
@jpetrell, @kananoja, please, review.

File: src/mprisplayer.cpp [hunk 1]
```diff
@@ -70,6 +70,8 @@ MprisPlayer::MprisPlayer(QObject *parent)
 
 MprisPlayer::~MprisPlayer()
 {
+    // Unregister upon destruction
+    setServiceName(QStringLiteral(""));
```
Comment (kananoja) [inline src/mprisplayer.cpp]: Hmm... Not sure this is the best way to unregister the service. setServiceName will call "registerService" which will print a warning if the service name is empty. The service will get unregistered, but I don't like the warning. Either split "unregisterService" as a separate function, or just copy-paste the couple of lines here.

(Also not nice to emit "serviceNameChanged" signal in the destructor.)

Comment (tanty) [inline src/mprisplayer.cpp]: Fair enough!

**Comment by tanty** [comment]
Updated. @kananoja , please, review! :)

**Comment by kananoja** [comment]
LGTM

**Pull request merged**